### PR TITLE
Adds size and availability filters and relevant tests

### PR DESCRIPTION
**Pull request by stevemac1900**

### CODE REVIEW

File: src/availability_filter.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AvailabilityFilter:
+    def __init__(self, subfilter, availability):
+        self.subfilter = subfilter
+        self.availability = availability
+
+    def is_match(self, beer):
+        if beer.is_available != self.availability:
```
Comment (bjcoleman) [inline src/availability_filter.py]: This should be done by quantity, not by adding a field to the beer object.

Comment (bjcoleman) [reply src/availability_filter.py]: I hadn't thought about allowing for True and False for `availability`, but this is a good idea.

Make sure there is a test for positive, zero *and* negative - I recall seeing negative quantities in the data.

File: tests/test_availability_filter.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from availability_filter import AvailabilityFilter
+from beer import Beer
+from yes_filter import YesFilter
+
+
+def test_name_filter_match():
```
Comment (bjcoleman) [inline tests/test_availability_filter.py]: You also need a test for not available.  And given my discussion in the filter class, you will need negative quantities.

File: tests/test_size_filter.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from size_filter import SizeFilter
+from beer import Beer
+from yes_filter import YesFilter
+
+
+def test_name_filter_match():
```
Comment (bjcoleman) [inline tests/test_size_filter.py]: You also need a test for false.

File: src/availability_filter.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class AvailabilityFilter:
+    def __init__(self, subfilter, quantity_available):
+        self.sf = subfilter
+        self.qa = quantity_available
```
Comment (bjcoleman) [inline src/availability_filter.py]: These class variable names are not Uncle Bob-compliant.

File: tests/test_availability_filter.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from availability_filter import AvailabilityFilter
+from beer import Beer
+from yes_filter import YesFilter
+
+
+def get_test_beer(quantity):
```
Comment (bjcoleman) [inline tests/test_availability_filter.py]: I like `make_test_beer` better than `get_test_beer`

Can/should this be a pytest fixture?

File: tests/test_availability_filter.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    return Beer("Guinness", "pint", "stout", 0, quantity)
+
+
+def get_test_filter(quantity):
```
Comment (bjcoleman) [inline tests/test_availability_filter.py]: same as above

**Comment by stevemac1900** [comment]
So with this change I don't check for the user requesting a zero value or negative values, I wasn't positive about this but I'm assuming that we're putting the onus of requesting positive values on the user.

**Comment by mtnesbitt** [comment]
Looks good

**Pull request merged**